examples/chef/common/chef-fan-control-manager.cpp

```diff
@@ -38,20 +38,34 @@ using namespace chip::app::Clusters::FanControl::Attributes;
 using Protocols::InteractionModel::Status;
 
 namespace {
-class ChefFanControlManager : public AttributeAccessInterface, public Delegate
+class ChefFanControlManager : public Delegate
 {
 public:
-    ChefFanControlManager(EndpointId aEndpointId) :
-        AttributeAccessInterface(Optional<EndpointId>(aEndpointId), FanControl::Id), Delegate(aEndpointId)
-    {}
+    ChefFanControlManager(EndpointId aEndpointId) : Delegate(aEndpointId) {}
 
-    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
-    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+    void Init();
+    void HandleFanControlAttributeChange(AttributeId attributeId, uint8_t type, uint16_t size, uint8_t * value);
     Status HandleStep(StepDirectionEnum aDirection, bool aWrap, bool aLowestOff) override;
+    DataModel::Nullable<uint8_t> GetSpeedSetting();
+    DataModel::Nullable<Percent> GetPercentSetting();
 
 private:
-    Nullable<uint8_t> mPercentSetting{};
-    Nullable<uint8_t> mSpeedSetting{};
+    uint8_t mPercentCurrent;
+    uint8_t mSpeedCurrent;
+
+    // Fan Mode Limits
+    static constexpr int FAN_MODE_LOW_LOWER_BOUND    = 1;
+    static constexpr int FAN_MODE_LOW_UPPER_BOUND    = 3;
+    static constexpr int FAN_MODE_MEDIUM_LOWER_BOUND = 4;
+    static constexpr int FAN_MODE_MEDIUM_UPPER_BOUND = 7;
+    static constexpr int FAN_MODE_HIGH_LOWER_BOUND   = 8;
+    static constexpr int FAN_MODE_HIGH_UPPER_BOUND   = 10;
```

Comment:
nit: we generally use `kCamelCase` for C++ constant naming

Reply:
I also wonder if we should generally have suffix for things .... there is no real unit of measurement here, but maybe `kFanModeLowSpeedLowerBound` (i.e. speed suffix) would make code obvious in `if (speed <= kFanModeLowSpeedLowerBound)`

Reply:
Maybe outside of this PR, but for maintainability, it seems that these are a `range` so I would want to have code like:

```cpp
constexpr Range kLowFanSpeed = {1, 3};
constexpr Range kMediumFanSpeed = {4, 7};
constexpr Range kHighFanSpeed = {8, 10};

static_assert(kLowFanSpeed.High() + 1 == kMediumFanSpeed.Low());
static_assert(kMediumFanSpeed.High() + 1 == kHighFanSpeed.Low());
// ....

if (kLowFanSpeed.Contains(speed))
{
    return FanControl::FanModeEnum::kLow;
}

// ...

if (!kLowFanSpeed.Contains(mSpeedCurrent))
{
    DataModel::Nullable<uint8_t> speedSetting(kLowFanSpeed.Low());
    // ...
```

Reply:
Good ideas, done all that! Thanks Andrei.